internal/command/command.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package command
+
+import (
+	"context"
+	"flag"
+	"fmt"
+
+	"github.com/googleapis/generator/internal/container"
+)
+
+type Command struct {
+	Name  string
+	Short string
+	Run   func(ctx context.Context) error
+
+	flags *flag.FlagSet
+}
+
+func (c *Command) Parse(args []string) error {
+	return c.flags.Parse(args)
+}
+
+func Lookup(name string) (*Command, error) {
+	var cmd *Command
+	for _, sub := range Commands {
+		if sub.Name == name {
+			cmd = sub
+		}
+	}
+	if cmd == nil {
+		return nil, fmt.Errorf("invalid command: %q", name)
+	}
+	return cmd, nil
+}
+
+var CmdConfigure = &Command{
+	Name:  "configure",
+	Short: "Configure a new API in a given language",
+	Run: func(ctx context.Context) error {
+		if flagAPIRoot == "" {
+			return fmt.Errorf("-api-root is not provided")
+		}
+		if flagAPIPath == "" {
+			return fmt.Errorf("-api-path is not provided")
+		}
+		if !supportedLanguages[flagLanguage] {
+			return fmt.Errorf("invalid -language flag specified: %q", flagLanguage)
+		}
+		if flagPush && flagGitHubToken == "" {
+			return fmt.Errorf("-github-token must be provided if -push is set to true")
+		}
+		return container.Configure(ctx, flagLanguage, flagAPIRoot, flagAPIPath, flagGeneratorInput)
+	},
+}
+
+var CmdGenerate = &Command{
+	Name:  "generate",
+	Short: "Generate a new client library",
+	Run: func(ctx context.Context) error {
+		if flagAPIRoot == "" {
+			return fmt.Errorf("-api-root is not provided")
+		}
+		if flagAPIPath == "" {
+			return fmt.Errorf("-api-path is not provided")
+		}
+		if !supportedLanguages[flagLanguage] {
+			return fmt.Errorf("invalid -language flag specified: %q", flagLanguage)
+		}
+		if _, err := cloneGoogleapis(ctx); err != nil {
```

Comment:
We only need to clone googleapis if api-root isn't specified (in which case, the location of the clone is the default)